Rocky9

pwalczysko opened the pull request:
- upgrade python 3.9, ubuntu 22.04 for running image
- fix references to ice packages in defaults
- bump omero-py
- change platforms to rocky9 and add ubuntu 22.04 platform and thus 22.04 test
- bump postgres to 14 (afai understand this is a desired version everywhere)

Inside the tasks/main.yml
- remove python3 setup (it is present by default)
- point to the config for omero-prometheus-tools directly under ``/etc`` on Ubuntu as the ``/etc/sys...`` dir does not exist there
- create a logic for the above in the ``main.yml``
- create a separate ``templates/systemd-prometheus-omero-exporter-service-debian.j2`` file with the correct path for Debian (i.e. ``/etc/prometheus-omero-exporter``)


Test is passing on https://github.com/pwalczysko/ansible-role-omero-prometheus-exporter/actions/runs/7561213705 

cc @jburel @khaledk2

sbesson commented on defaults/main.yml:
```diff
@@ -33,15 +33,15 @@ omero_prometheus_exporter_counts_query_files:
   - "{{ omero_prometheus_tools_requirements_ice_package[ansible_os_family] |
         default('zeroc-ice')}}"
-  - omero-py>=5.13.0
+  - omero-py>=5.18.0
```
Is this a requirement for the functionality of the role?

jburel replied:
I believe it is not but since we are moving to platform with Python 3.8+ as the default. I think it makes sense to bump

pwalczysko replied:
I will leave as it is now then.

sbesson commented on templates/systemd-prometheus-omero-exporter-service-debian.j2:
```diff
@@ -0,0 +1,13 @@
+
+[Service]
+User={{ omero_prometheus_exporter_system_user }}
+EnvironmentFile=-/etc/prometheus-omero-exporter
```
Since these are templates already, should the value of `EnvironmentFile` become a variable itself?

`defaults/main.yml` could define a dictionary of the values of this environment file depending on the operating system family

pwalczysko replied:
fixed in https://github.com/ome/ansible-role-omero-prometheus-exporter/pull/14/commits/8df2fe974d6f0ea28a3d2917cb7c1c3b0ba38d03

pwalczysko commented:
The tests are green. I have 

- fixed the OS-dependent logic by introducing a new variable cf. @sbesson 's comment
- reverted the minimal ansible version bump as discussed with @jburel on other PRs

Ready for review.

sbesson commented on tasks/main.yml:
```diff
@@ -65,6 +49,20 @@
+- name: omero prometheus exporter | configure service
+  become: true
+  template:
+    dest: /etc/prometheus-omero-exporter
```
Note that this task could also been combined with the previous ones to use `environment_file_path[ansible_os_family]`. 

Functionally, there should be no functional difference between using this style, `when` in the tasks or even splitting OS-specific tasks into separate subplaybooks e.g. as in https://github.com/ome/ansible-role-postgresql-client. My only recommendation would be to reduce the variations to facilitate your maintenance moving forward

sbesson reviewed (approved):
Barring the point about handling OS-specific tasks in https://github.com/ome/ansible-role-omero-prometheus-exporter/pull/14#discussion_r1457626669 which is really a decision for the @ome/devops team, the proposed changes look good.

Once published, I can update https://github.com/IDR/deployment/pull/413 to deploy the newest version role to the pilot VM to be deployed on Rocky Linux 9.

pwalczysko commented:
@sbesson 

> Note that this task could also been combined with the previous ones to use `environment_file_path[ansible_os_family]`.

Sorry, simply went with your suggestion as:
1. I like it myself
2. It simplifies and shortens the code and thus facilitates maintenance
3. Do not see the reason for the ``when`` approach as just one line of difference between OS-es here is different from the cited postgresql-client case
4. I believe @jburel and, most probably @khaledk2 would agree with this approach - basically, have the cleanest code possible ?

cc @jburel @khaledk2 

(it will need one more approving review please, as GitHub claims that I have "dismissed" the previous review by the new commit :)

sbesson commented on meta/main.yml:
```diff
@@ -4,11 +4,14 @@ galaxy_info:
   company: Open Microscopy Environment
   license: BSD
-  min_ansible_version: 2.6
+  min_ansible_version: 2.3
```
Just spotted this. Is this an oversight?

jburel replied:
I think @pwalczysko got confused with a previous comment on a PR where the ``min_ansible_version`` was set to 2.3 and he bumped it to 2.10
2.10 is required when we have dependencies otherwise we do not have any reason to change it unless we can to set to 2.10 across the various repositories

pwalczysko replied:
@sbesson yes, oversignt, thank you. @jburel just clarified - if we have dependencies (there are some here in the requirements file), we are going for 2.10.

fixed in https://github.com/ome/ansible-role-omero-prometheus-exporter/pull/14/commits/74cba3446061ce8b7e416cbfaacd819525fc2151

sbesson reviewed (changes requested):
One last minute issue spotted with the Ansible version declared in `meta/main.yml`.

The latest state of your PR brings multi-platform support while increasing the complexity of the role and/or adding redundant blocks. No objection to moving forward with this change from my side

pwalczysko commented:
Test is green. Thanks @sbesson @jburel . Merging.
